lib/SparkPost/SparkPost.php

```diff
@@ -64,11 +65,16 @@ public function setupUnwrapped ($endpoint) {
    * Merges passed in headers with default headers for http requests
    */
   public function getHttpHeaders() {
+
     $defaultOptions = [
         'Authorization' => $this->config['key'],
         'Content-Type' => 'application/json',
     ];
 
+    if( $this->config['subaccount_id'] ){
```

Comment:
Hmm, I would think that instantiating a new class is a cleaner approach rather than switching sub id's. Besides, if you are going to use the apikey created for a sub account you are going to have to call setConfig anyways to change the key, right? 

From my experience the X-MSYS-SUBACCOUNT header is basically ignored on things like inbound-domains and other non subaccount api resources. 

Obviously just my opinions. Educate me ;-)